kratos/includes/kernel.h

```diff
@@ -141,10 +145,12 @@ class KRATOS_API(KRATOS_CORE) Kernel {
     ///@name Static Member Variables
     ///@{
 
-        KratosApplication::Pointer mpKratosCoreApplication;
+    KratosApplication::Pointer mpKratosCoreApplication;
 
     static bool mIsDistributedRun;
 
+    static int mNumThreads;
```

Comment:
@pooyan-dadvand as this also touches the `Kernel` can you please have a look?

Reply:
I agree with @RiccardoRossi about having it stored in a common place but I am not sure if the kernel is the natural place for it. I think it is more natural to be stored in the same parallel helpers, or the parallel utilities. Nevertheless, I don't have a strong opinion about it and open to keep it here if others feel that it is more coherent to be here. 

Implementation wise I would go for a complete singleton implementation rather than a simple static member function. Please note that in this implementation if someone uses this variable before setting, the result would be undefined. A better approach would be a simple Meyer's singleton. The best approach however would be the complete singleton with destructor ward as @jcotela implemented in the Parallel environment.